### DISCUSSION TITLE
Interface configuration mode

### DISCUSSION THREAD

**Discussion by seimnseimn** [scrapli/scrapli_community#172] [Q&A] [answered]
Dear all, I'm new to Scrapli and so far I guess I understand the basics.

_This might or might not be specific to huawei/smartax._

When PRIVILEGE_LEVELS are defined properly I can send configurations e.g. with

```python
async with AsyncScrapli(**my_device) as conn:
    result = await conn.send_config(
        'some config line',
        privilege_level="configuration",
    )
```

however if I need to enter an interface first, then do a config there and "leave" the interface again. Is there a best practice or a predefined "flow" how to do this? Do I need to create another privilege level or how would I attempt this properly?

I could of course do:

```python
results = await conn.send_configs([
    "interface gpon 0/1",
    "some config on the interface",
    "quit"],
    privilege_level="configuration"
)
```

but then what if "interface gpon 0/1" does not exist, then the 3rd line "quit" would make me leave the configuration mode. And also the parsing of the "some config line" seems a bit difficult.

Maybe I missed something in the docs. Any help is appreciated :-)

**Comment by carlmontanari** [comment, marked as answer]
hey @seimnseimn 

what you've outlined here is what I would do (run the commands in the interface then "quit"). scrapli doesn't really care about/understand the config/structure of the router/switch/thing so its up to you to decide how to handle things. if you were worried about the interface not existing you could always do some show command then conditionally run your config based on that output. 

you *could* probably get nutty with adding privilege levels and stuff but I dont think id advise that for interfaces or anything like that.

**Reply by seimnseimn** [reply]
Thank you @carlmontanari

just wanted to to make sure that I'm not doing something wrong. As you already mentioned, adding privilege levels might be confusing or even impossible as the escalation to that privilege level would actually require a (or multiple) parameter(s).

Let me get familiar a bit more with scrapli/-community and probably I will ad a few small PR's regarding smartax
